src/AnalyzerVisitors/IrradiationPower.cc

```diff
@@ -67,31 +67,44 @@ void IrradiationPowerVisitor::visit(DetectorModule& m) {
   // counter
   modulesCounter_[moduleRef]++;
   // The list of modules per irradiation type
-  mapTypeToIrradiation_[modType].push_back(irradiationMean);
+  mapTypeToIrradiation_[modType].push_back(&m);
 }
 
 
 void IrradiationPowerVisitor::postVisit() {
   // Sort the irradiation and extract the max and 95% percentiles
   sensorsIrradiationPerType.setCell(0, 0, "Type");
   sensorsIrradiationPerType.setCell(0, 1, "# mods");
-  sensorsIrradiationPerType.setCell(0, 2, "Max. module average irrad [Hb]");
-  sensorsIrradiationPerType.setCell(0, 3, "95 percentile module average irrad [Hb]");
+  sensorsIrradiationPerType.setCell(0, 2, "Max. module irrad (avg) [Hb]");
+  sensorsIrradiationPerType.setCell(0, 3, "95% module irrad (avg) [Hb]");
+  sensorsIrradiationPerType.setCell(0, 4, "z_max [mm]");
+  sensorsIrradiationPerType.setCell(0, 5, "r_max [mm]");
   int iRow=0;
   for (auto& it : mapTypeToIrradiation_ ) {
     iRow++;
     const std::string& typeName = it.first;
-    std::vector<double>& irrads = it.second;
-    std::sort(irrads.begin(), irrads.end());
+    std::vector<const DetectorModule*>& irrads = it.second;
```

Comment:
irrads : 
Why not using a std::set here ?
Also, why not simply having double instead of DetectorModule ?